.github/workflows/release.yml

```
@@ -13,49 +13,61 @@ jobs:
     steps:
       - name: Checkout commit
         uses: actions/checkout@v2
+      - name: Get Tag name
+        uses: little-core-labs/get-git-tag@v3.0.1
+        id: tagName
+        with:
+          tagRegex: "BertModel-(\\d+).*"
       - name: Setup Python 3.8
         uses: actions/setup-python@v2
         with:
           python-version: '3.8'
       - name: Configure GCP access
-        uses: google-github-actions/setup-gcloud@master
-        with:
-          project_id: $ {{ secrets.GCP_PROJECT_NAME }}
-          service_account_key: ${{ secrets.GCP_CREDS_JSON_BASE64 }}
-          export_default_credentials: true
-      - name: Alternative for GCP access
         uses: actions-hub/gcloud@master
         env:
-          PROJECT_ID: $ {{ secrets.GCP_PROJECT_NAME }}
+          PROJECT_ID: engineeringlab
           APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
         with:
           args: info
-      # DockerHub integration via actions
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
-      - name: Login to DockerHub
-        uses: docker/login-action@v1 
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ steps.tagName.outputs.tag }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Configure Docker
+        uses: actions-hub/gcloud@master
+        env:
+          PROJECT_ID: engineeringlab
+          APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+        with:
+          args: auth configure-docker gcr.io --quiet
+      - name: Login to GCR
+        uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          registry: gcr.io
+          username: _json_key
+          password: ${{ secrets.GCR_JSON_KEY }}
       - name: Build and push
-        id: docker_build
         uses: docker/build-push-action@v2
         with:
-          push: true
           context: ./Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier
-          file: ./Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile
-          tags: englabsteam1/trainer:latest
+          file: ./Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-torchserve
+          push: true
+          tags: gcr.io/engineeringlab/torchserve:${{ steps.tagName.outputs.tag }}
```

For possible different models
```suggestion
          tags: gcr.io/engineeringlab/torchserve:BertModel-${{ steps.tagName.outputs.tag }}
```

---

Or you could just change the regex to capture the whole group, `tagRegex: "(BertModel-\\d+).*"`?

---

Done